pallets/asset-index/src/lib.rs

```diff
@@ -665,6 +712,71 @@ pub mod pallet {
 			share.checked_div(&nav).ok_or_else(|| ArithmeticError::Overflow.into())
 		}
 
+		/// Consolidate and clean deposits while completing withdraw
+		fn do_consolidate_deposits(
```

Comment:
We need to this in `withdraw`, basically the workflow is:
1. call `withdraw` with `amount` of pint
2. subtract base fee (we already do this)
3. now compute the redemption fee and subtract from `amount` as well, this is basically this `consolidition`
   in order to do that we need to iterate over deposits and substract the single `Deposits` from `amount` and accumulate the fees.